frontend/src/metabase/dashboard/containers/DashboardApp/DashboardApp.tsx

```diff
@@ -171,24 +188,52 @@ const DashboardApp = (props: DashboardAppProps) => {
     onTimeout,
   });
 
+  const { refreshDashboard } = useRefreshDashboard({
+    dashboardId: dashboardId,
+    parameterQueryParams,
+  });
+
   const {
-    documentTitle: _documentTitle,
-    pageFavicon: _pageFavicon,
-    isRunning: _isRunning,
-    isLoadingComplete: _isLoadingComplete,
-    children,
-    ...dashboardProps
-  } = props;
+    hasNightModeToggle,
+    hideDownloadButton,
+    hideParameters,
+    isFullscreen,
+    isNightMode,
+    onNightModeChange,
+    refreshPeriod,
+    onFullscreenChange,
+    setRefreshElapsedHook,
+    onRefreshPeriodChange,
+    theme,
+    titled,
+    font,
+  } = useDashboardUrlParams({ location, onRefresh: refreshDashboard });
+
+  useSyncURLSlug({ location });
 
   return (
     <div className={cx(CS.shrinkBelowContentSize, CS.fullHeight)}>
       <LeaveConfirmationModal isEnabled={isEditing && isDirty} route={route} />
-
-      {/* @ts-expect-error for now until we can get the prop-drilled types sorted out. Previously DashboardControls was a JS file so types weren't checked, but now there's a Pandora's box here */}
+      {/* @ts-expect-error - types coming from thunks are not correct*/}
```

Comment:
This was too big to try and fix in this PR, I'll see in the next days, I may have an idea to do it relatively quickly